src/blr/cli/clusterrmdup.py

```diff
@@ -3,594 +3,303 @@
-            pass
-
-    def reportMergeDict(self, merge_dict):
-        """ Saves a readable string format of merge_dict to write to out."""
-
-        self.ClustersRemovedDueToMerge = len(merge_dict.keys())
-
-    def coupling_analysis(self):
-        """ Summarises overlap into a a coupling analysis dictionary (~bins values)."""
-
-        # Iterate over 'droplet defining' clusterids
-        for lower_barcode in self.overlap_dict.keys():
-            barcodes_in_drop = len(self.overlap_dict[lower_barcode].keys())
-
-            # iterates over translation event and counts # of puts into key according to how many duplicate reads that event had
-            for translation_events, number_of_reads in self.overlap_dict[lower_barcode].items():
-                try: self.coupling_dict[barcodes_in_drop][number_of_reads] += 1
-                except KeyError:
-                    try: self.coupling_dict[barcodes_in_drop]
-                    except KeyError:
-                        self.coupling_dict[barcodes_in_drop] = dict()
-                    self.coupling_dict[barcodes_in_drop][number_of_reads] = 1
-
-        self.readable_coupling_dict += 'bc/drop\toverlap\t#occurances\n'
-        for number_of_bc_in_droplet in sorted(self.coupling_dict.keys()):
-
-            for number_of_overlapping_reads in sorted(self.coupling_dict[number_of_bc_in_droplet].keys()):
-
-                self.readable_coupling_dict += str(number_of_bc_in_droplet) + '\t' + str(number_of_overlapping_reads) + '\t' + str(self.coupling_dict[number_of_bc_in_droplet][number_of_overlapping_reads]) + '\n'
-                self.readable_coupling_dict += str(number_of_bc_in_droplet) + '\t' + str(number_of_overlapping_reads) + '\t' + str(self.coupling_dict[number_of_bc_in_droplet][number_of_overlapping_reads]) + '\n'
-
-    def writeLog(self):
+        # Read stats
+        logger.info(f"Total Reads in file: {self.tot_reads}")
+        logger.info(f"  Unmapped reads: {self.unmapped_reads}")
+        logger.info(f"  Reads with unmapped mate: {self.unmapped_mates}")
+        logger.info(f"  Reads without {barcode_tag} tag: {self.non_tagged_reads}")
+        logger.info(f"  Reads analyzed: {self.reads_analyzed}")
+        logger.info(f"Reads at dup pos: {self.reads_at_analyzed_dup_position}")
+        logger.info(f"Reads with new tag: {self.reads_with_new_tag}")
+        logger.info(f"{barcode_tag} tags removed: {self.barcodes_removed}")
 
-        import time
-        with open(self.log, 'a') as openout:
-            openout.write(time.strftime("%a, %d %b %Y %H:%M:%S", time.localtime()))
-            for objectVariable, value in vars(self).items():
-                openout.write('\n\n'+str(objectVariable) + '\n' + str(value))
 
 def add_arguments(parser):
     parser.add_argument("input_tagged_bam", help=".bam file tagged with RG tags and duplicates marked (not taking "
                                                  "cluster id into account).")
```

Comment:
Great catch, just writing it here for future documentation that the files does not need to be indexed anymore since the way of calculating summary stats was switched, so I will leave that part out.